### PR TITLE
Port/manual assignment

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Port manual assignment
- Nether fix
- Creative builder inventory clicking

From 1.10 to 1.11

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java [hunk 1]
```diff
@@ -96,7 +97,7 @@ public EntityAIStructureMiner(@NotNull final JobMiner job)
     private static boolean isOre(final Block block)
     {
         //TODO make this more sophisticated
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java:99]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 84 issues
* ![MAJOR][MAJOR] 35 major
* ![MINOR][MINOR] 46 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.273 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): class "Colony" has 86 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L202](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L202): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L601](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L601): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1084](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1084): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1085](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1085): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1209](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1209): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1211](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1211): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1213](https://github.com/Minecolonies/minecolonies/blob/9393b9576e6932bee4f31d5a43381128ace734d6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1213): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 84 issues
* ![MAJOR][MAJOR] 35 major
* ![MINOR][MINOR] 46 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.274 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): class "Colony" has 86 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L202](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L202): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L601](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L601): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1085](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1085): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1086](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1086): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1210](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1210): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1212](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1212): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1214](https://github.com/Minecolonies/minecolonies/blob/20daefcc501bb774c751434dcd65bdac2f0f8db0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1214): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by xavierh — APPROVED** [review]

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 84 issues
* ![MAJOR][MAJOR] 35 major
* ![MINOR][MINOR] 46 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.274 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): class "Colony" has 86 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L202](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L202): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L601](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L601): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1085](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1085): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1086](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1086): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1210](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1210): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1212](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1212): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1214](https://github.com/Minecolonies/minecolonies/blob/6e7ad0e3591277426dd3227e165620d372a5b24d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1214): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**